R/nested_resampling.R

```diff
@@ -0,0 +1,191 @@
+#' over the [AutoTuner] objects and binds the tuning archives to a
+#' [data.table::data.table()]. [AutoTuner] must be initialized with
+#' `store_tuning_instance = TRUE` and `resample()` or `benchmark()` must be
+#' called with `store_models = TRUE`.
+#'
+#' @section Data structure:
+#'
+#' The returned data table has the following columns:
+#'
+#' * `experiment` (integer(1))\cr
+#'   Experiment of the benchmark grid.
+#' * `iteration` (integer(1))\cr
+#'   Iteration of the outer resampling.
+#' * One column for each hyperparameter of the search spaces.
+#' * One column for each performance measure.
+#' * `runtime` (`numeric(1)`)\cr
+#'   Sum of all training and predict times logged per resampling.
+#' * `timestamp` (`POSIXct`)\cr
+#'   Time stamp when the evaluation was logged into the archive.
+#' * `batch_nr` (`integer(1)`)\cr
+#'   Hyperparameters are evaluated in batches. Each batch has a unique batch
+#'   number.
+#' * `x_domain` (`list()`)\cr
+#'   List of transformed hyperparameter values. By default this column is
+#'   unnested.
+#' * `x_domain_*` (`any`)\cr
+#'   Separate column for each transformed hyperparameter.
+#' * `resample_result` ([mlr3::ResampleResult])\cr
+#'   Resample result of the inner resampling.
+#' * `task_id` (`character(1)`).
+#' * `learner_id` (`character(1)`).
+#' * `resampling_id` (`character(1)`).
+#'
+#' @param x ([mlr3::ResampleResult] | [mlr3::BenchmarkResult]).
+#' @param unnest (`character()`)\cr
+#'   Transforms list columns to separate columns. By default, `x_domain` is
+#'   unnested. Set to `NULL` if no column should be unnested.
+#' @param exclude_columns (`character()`)\cr
+#'   Exclude columns from result table. Set to `NULL` if no column should be
+#'   excluded.
+#' @return [data.table::data.table()].
+#'
+#' @export
+#' @examples
+#' task = tsk("iris")
+#' learner = lrn("classif.rpart", cp = to_tune(0.001, 0.1))
+#'
+#' at = AutoTuner$new(
+#'   learner = learner,
+#'   resampling = rsmp("holdout"),
+#'   measure = msr("classif.ce"),
+#'   terminator = trm("evals", n_evals = 5),
+#'   tuner = tnr("grid_search"),
+#'   store_tuning_instance = TRUE)
+#'
+#' resampling_outer = rsmp("cv", folds = 3)
+#' rr = resample(task, at, resampling_outer, store_models = TRUE)
+#'
+#' extract_inner_tuning_archives(rr)
+extract_inner_tuning_archives = function (x, unnest = "x_domain", exclude_columns = "uhash") {
+   UseMethod("extract_inner_tuning_archives")
+}
+
+#' @export
+extract_inner_tuning_archives.ResampleResult = function(x, unnest = "x_domain", exclude_columns = "uhash") {
+  rr = assert_resample_result(x)
+  if (is.null(rr$learners[[1]]$model) || is.null(rr$learners[[1]]$model$tuning_instance)) {
+    return(data.table())
+  }
+  tab = imap_dtr(rr$learners, function(learner, i) {
+    data = as.data.table(learner$archive, unnest, exclude_columns)
+    set(data, j = "iteration", value = i)
+  })
+  tab[, "task_id" := rr$task$id]
```

Comment:
Could be done [in one line](http://brooksandrew.github.io/simpleblog/articles/advanced-data-table/#create-multiple-columns-with--in-one-statement). Maybe you already knew. Not saying that this should be done here. Just FYI

Reply:
No I didn't know that. Thanks!